runtime/lua/vim/lsp/util.lua

```diff
@@ -1353,6 +1353,39 @@ function M.character_offset(buf, row, col)
   return str_utfindex(line, col)
 end
 
+-- for use with `formatexpr` if called without parms
+-- @param start_line 1-indexed line
+-- @param end_line 1-indexed line
+-- @param timeout_ms optional
+function M.formatexpr(start_line, end_line, timeout_ms)
+  if not start_line or not end_line then
+    if vim.fn.mode() == 'i' or vim.fn.mode() == 'R' then
+      -- `formatexpr` is also called when exceding
+      -- `textwidth` in insert mode
+      -- fall back to internal formatting
+      return 1
+    end
+    start_line = vim.v.lnum
+    end_line = start_line + vim.v.count - 1
+  end
+  if start_line > 0 and end_line > 0 then
+    local params = {
+      textDocument = { uri = vim.uri_from_bufnr(0) };
+      range = {
+        start = { line = start_line - 1; character = 0; };
+        ["end"] = { line = end_line - 1; character = 0; };
+      };
+    };
+    local result = vim.lsp.buf_request_sync(0, "textDocument/rangeFormatting", params, timeout_ms)
+    if result then
+      result = result[1].result
+      vim.lsp.util.apply_text_edits(result)
+    end
```

Comment:
Should this return 1 if the LSP did not have a result?

Will the result always be a list?

Can you not re-set the value `result` to be something different from what it is originally?

Reply:
Sorry, I wasn't super clear.

I was asking about the structure of the result (like, could it return a single item, rather than a list of items). As in, the result we get from the LSP.

Reply:
As i understand it, it returns a map with the result from each running LSP or `(nil,error)` on failure